django/db/models/sql/query.py

```diff
@@ -73,6 +73,19 @@ def get_children_from_q(q):
             yield child
 
 
+def rename_prefix_from_q(prefix, replacement, q):
```

Comment:
This would need to be handle expressions as well but it should be relatively trivial to implement; basically all refs to `relation__` must be turned into `alias__` instead.